lib/prometheus_exporter/instrumentation/active_record.rb

```diff
@@ -7,19 +7,19 @@ class ActiveRecord
 
     def self.start(client: nil, frequency: 30, custom_labels: {}, config_labels: [])
 
-      # Not all rails versions support coonection pool stats
-      unless ::ActiveRecord::Base.connection_pool.respond_to?(:stat)
-        STDERR.puts("ActiveRecord connection pool stats not supported in your rails version")
-        return
-      end
-
       config_labels.map!(&:to_sym)
       validate_config_labels(config_labels)
 
       active_record_collector = new(custom_labels, config_labels)
 
       client ||= PrometheusExporter::Client.default
 
+      # Not all rails versions support connection pool stats
+      unless ::ActiveRecord::Base.connection_pool.respond_to?(:stat)
+        client.logger.error("ActiveRecord connection pool stats not supported in your rails version")
+        return
+      end
+
```

Comment:
Are we sure the other stuff is supported here on all Rails versions, I worry that this fix may raise even more support issues when it is partially collecting?

Reply:
I'm not sure I follow; the only change here is I moved this code from a few lines higher down to here that way we could use `client.logger`.

I looked at the code that is now executing (that wouldn't have prior to my change) and it looks like it's all internal setup for this gem and not dependent on Rails at all. Code that might now execute before the guard clause instead of after:
```ruby
      config_labels.map!(&:to_sym) # harmless
      validate_config_labels(config_labels) # internal validation. this could now raise whereas before it wouldn't even execute

      active_record_collector = new(custom_labels, config_labels) # assigns ivars, harmless

      client ||= PrometheusExporter::Client.default # instantiates client but doesn't actually do any work, harmless
```

The `raise` could be a problem if an app that doesn't support `stat` is _also_ passing in invalid config labels as now that app would `raise`. I'll move that code further down so it preserves the current behavior.

Reply:
This should better preserve the existing behavior: [`03a3714` (#177)](https://github.com/discourse/prometheus_exporter/pull/177/commits/03a37145fb1694b487bf7b79d4befc4189357f34)